pkg/controllers/bigtableautoscaler_controller.go

```diff
@@ -79,29 +79,36 @@ func (r *BigtableAutoscalerReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 	ctx := context.Background()
 	r.clock = clock.RealClock{}
 
-	autoscaler, err := r.getAutoscaler(req.NamespacedName)
+	autoscaler, err := r.getAutoscaler(ctx, req.NamespacedName)
+	clusterRef := autoscaler.Spec.BigtableClusterRef
 
 	if err != nil {
```

Comment:
```suggestion
	clusterRef := autoscaler.Spec.BigtableClusterRef
	autoscaler, err := r.getAutoscaler(ctx, req.NamespacedName)
	if err != nil {
```
I think it is a good practice in go to do the err check just after its attribution.